project-addons/custom_partner/partner.py

```diff
@@ -764,6 +764,38 @@ def compute(self, period, invoice_lines, refund_lines, tmp_model=False):
         return True
 
 
+class rappel(models.Model):
+
+    _inherit = 'rappel'
+    brand_ids = fields.Many2many('product.brand', 'rappel_product_brand_rel',
+                                 'rappel_id', 'product_brand_id', 'Brand')
+
+    @api.multi
+    def get_products(self):
+        product_obj = self.env['product.product']
+        product_ids = self.env['product.product']
+        for rappel in self:
+            if not rappel.global_application:
+                if rappel.product_id:
+                    product_ids += rappel.product_id
+                elif rappel.brand_ids:
+                    product_ids += product_obj.search(
+                        [('product_brand_id', 'in', rappel.brand_ids.ids)])
+                elif rappel.product_categ_id:
+                    product_ids += product_obj.search(
+                        [('categ_id', '=', rappel.product_categ_id.id)])
+            else:
+                product_ids += product_obj.search([])
+        return [x.id for x in product_ids]
```

Comment:
product_ids.ids

Reply:
Pero el id lo saco con x.id al hacer el for

Reply:
no hace falta el for "[x.id for x in product_ids]" es lo mismo que "product_ids.ids"